File: solutions/cpp/space-age/1/space_age.cpp
#include "space_age.h"

namespace space_age {
    enum Factors : uint64_t {
        earth = 31557600,
        mercury = 24084670,
        venus = 61519726,
        mars = 188081580,
        jupiter = 1186261500,
        saturn = 2944749800,
        uranus = 8401684600,
        neptune = 16479132000,
    };

    static const double factor = 1e8;

    uint64_t space_age::seconds() const { return age_; }
    double space_age::on_earth() const { return static_cast<double>(seconds()) / earth; }
    double space_age::on_mercury() const { return on_earth() / mercury * factor; }
    double space_age::on_venus() const { return on_earth() / venus * factor; }
    double space_age::on_mars() const { return on_earth() / mars * factor; }
    double space_age::on_jupiter() const { return on_earth() / jupiter * factor; }
    double space_age::on_saturn() const { return on_earth() / saturn * factor; }
    double space_age::on_uranus() const { return on_earth() / uranus * factor; }
    double space_age::on_neptune() const { return on_earth() / neptune * factor; }
}  // namespace space_age

File: solutions/lua/space-age/1/space-age.lua
local SpaceAge = {}

function SpaceAge:new(seconds)
  local on_earth = seconds / (60 * 60 * 24 * 365.25)
  local round2 = function(x) return tonumber(("%.2f"):format(x)) end
  self.seconds = seconds
  self.on_earth = function() return round2(on_earth) end
  self.on_mercury = function() return round2(on_earth / 0.2408467) end
  self.on_venus = function() return round2(on_earth / 0.61519726) end
  self.on_mars = function() return round2(on_earth / 1.8808158) end
  self.on_jupiter = function() return round2(on_earth / 11.862615) end
  self.on_saturn = function() return round2(on_earth / 29.447498) end
  self.on_uranus = function() return round2(on_earth / 84.016846) end
  self.on_neptune = function() return round2(on_earth / 164.79132) end
  return self
end

return SpaceAge
